SpatialGDK/Source/SpatialGDK/Private/Utils/LatencyManager.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/LatencyManager.h"
+
+#include "Engine/World.h"
+
+#include "EngineClasses/SpatialNetConnection.h"
+#include "EngineClasses/SpatialNetDriver.h"
+#include "GameFramework/PlayerController.h"
+#include "GameFramework/PlayerState.h"
+#include "Interop/Connection/SpatialWorkerConnection.h"
+#include "Interop/SpatialReceiver.h"
+#include "SpatialConstants.h"
+
+LatencyManager::LatencyManager(const USpatialNetConnection& InConnection, const USpatialNetDriver& InDriver)
+	: PlayerControllerEntity(SpatialConstants::INVALID_ENTITY_ID)
+	, NetConnection(InConnection)
+	, NetDriver(InDriver)
+{
+}
+
+void LatencyManager::Enable(Worker_EntityId InPlayerControllerEntity)
+{
+	checkf(PlayerControllerEntity == SpatialConstants::INVALID_ENTITY_ID, TEXT("LatencyManager::Enable : PlayerControllerEntity already set: %lld. New entity: %lld"), PlayerControllerEntity, InPlayerControllerEntity);
+	PlayerControllerEntity = InPlayerControllerEntity;
+
+	LastPingSent = NetConnection.GetWorld()->RealTimeSeconds;
+
+	auto Delegate = TBaseDelegate<void, const Worker_ComponentUpdateOp &>::CreateLambda([this](const Worker_ComponentUpdateOp& Op)
+	{
+		float ReceivedTimestamp = NetConnection.GetWorld()->RealTimeSeconds;
+
+		Schema_Object* EventsObject = Schema_GetComponentUpdateEvents(Op.update.schema_type);
+		uint32 EventCount = Schema_GetObjectCount(EventsObject, SpatialConstants::PING_PONG_EVENT_ID);
+
+		if (EventCount > 0)
+		{
+			NetConnection.PlayerController->PlayerState->UpdatePing(ReceivedTimestamp - LastPingSent);
+			SendPingOrPong(NetDriver.IsServer() ? SpatialConstants::SERVER_PING_COMPONENT_ID : SpatialConstants::CLIENT_PONG_COMPONENT_ID);
+		}
+	});
+
+
+	if (NetDriver.IsServer())
+	{
+		NetDriver.Receiver->AddClientPongDelegate(PlayerControllerEntity, Delegate);
```

Comment:
What are the usecases this manager would get re-used with a new entity id? Should the old delegate be removed?